### DISCUSSION TITLE
PINNs Inverse Problem - Exponential term

### DISCUSSION THREAD

**Discussion by KostasVog20** [lululxvi/deepxde#1639] [General]
Hi to everyone!

I want to use the DeepXDE package for an Inverse Problem (with the purpose to identify two constants in my equations). I have a system of 3 coupled equations, one PDE and two ODEs, with three dependent variables (T, S1, S2). The first equation has an exponential term in the form of : exp( - b / T ) where b is a constant and T is one of the above dependent variables.  When I insert a term in my code such as: exp ( - b / y[: , 0:1] ) I take a NaN in the Test Loss and the code stops. Can I have such terms in my equations or what can I do to solve it?

**Comment by haison19952013** [comment]
try to add a small value to T to avoid infinity, perhaps ? So, i guess it should be `exp ( - b / (y[: , 0:1] + 1e-6) )`. If this does not solve your problem, plz try to post your toy code here as well as the errors for us to investigate together.

**Comment by KostasVog20** [comment]
Thank you very much for your quick response!

I tried your suggestion, but it doesn't solve my problem. What I did, was that I changed the activation function from "tanh" to "relu" with the purpose to taking only positive results from the neurons. The problem probably was that temperature has taken negative values and the exponential term produced this error. Now, I have three more questions:
1) Can we set this "relu" function only to the output neurons and give the chance to the rest of the neurons to have the "tanh" activation (for the robustness of the system)? 
2) I can't understand the meaning of the variables num_domain, num_boundary, num_initial.
3) Where can I found the meaning of the columns of the train loss. I have 12 columns but I can't understand their meaning.

**Reply by haison19952013** [reply]
- How do you set the activation function for the output layer? As I remember, you can not set the activation for the last layer. The Linear activation function is automatically set for the last layer
1. You just need to set `tanh` as usual and use `apply_output_transform` to apply the `relu` for the output. In my opinion, it's might better to use `softplus` for the output in case your output is restricted as positive
2. num_domain, num_boundary, num_initial: number of residual points generated inside the domain, on the boundary, and at t = 0, respectively
3. For example, if you set up the `data` as follows
```
data = dde.data.PDE(geom,pde,[bc_inner, bc_outer],...)
```
The order of the losses will be losses from `pde`, losses from `bc_inner`, and losses from `bc_outer`

**Reply by KostasVog20** [reply]
@haison19952013 @lululxvi  if you could answer my questions there (https://github.com/lululxvi/deepxde/discussions/1644) would be grateful. I started a new general discussion to explain the problem.
